### PR TITLE
DB-11116 Allow parameter in unary cast and extract functions

### PR DESCRIPTION
**Pull request by ascend1**

### CODE REVIEW

**Comment by ascend1** [comment]
jenkins please test branch @cdh6.3.0

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2246/

**Comment by ascend1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2247/

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by msirek — APPROVED** [review]

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/CastNode.java [hunk 1]
```diff
@@ -1030,47 +1030,58 @@ private void genDataValueConversion(ExpressionClassBuilder acb,
...
-            } else if(srcTypeId.isDateTimeTimeStampTypeId() && opndType.getJDBCTypeId() == Types.TIMESTAMP) {
-                try {
-                    length = SQLTimestamp.getFormatLength(getCompilerContext().getTimestampFormat());
-                } catch(IllegalArgumentException e)
-                {
-                    // we shouldn't get here as GenericStatement should've checked that, but let's be defensive
-                    // and not throw
-                    length = getCompilerContext().getTimestampFormat().length() + 10;
-                }
+                    length = 254;
```
Comment (jyuanca) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/CastNode.java:1045]: Can you add comments or define constant for magic numbers?

**Review by jyuanca — APPROVED** [review]
You can address the minor comments in a follow-up Jira

**Review by carolp-503 — APPROVED** [review]
+1

**Pull request merged**